graph/src/data/store/mod.rs

```diff
@@ -563,11 +564,15 @@ impl Entity {
         v
     }
 
-    /// Try to get this entity's ID
+    /// Return the ID of this entity. If the ID is a string, return the
+    /// string. If it is `Bytes`, return it as a hex string with a `0x`
+    /// prefix. If the ID is not set or anything but a `String` or `Bytes`,
+    /// return an error
     pub fn id(&self) -> Result<String, Error> {
```

Comment:
Where is this `id` function used? Depending on where it's used, I'm curious if changing the return type of this to `Bytes` instead of `String` wouldn't help with performance too.

Reply:
We can't change it to `Bytes` because some entities might have actual string ids that can't be converted to `Bytes`. It's used in a few places where entity ids are compared (e.g., for the keys in the `EntityCache`) and IIRC for the explicit id that gets passed in from `store_set` (a string)

It would be a bit cleaner to make this an `enum` but that would require changing _a lot_ of code since it will ripple to `EntityKey`. I don't think it will buy us much since for any given entity type, all ids are either strings or bytes, and there's therefore no danger that a `String` id and a `Bytes` id might inadvertently clash.